src/mono/mono/mini/aot-runtime.c

```diff
@@ -1066,6 +1066,15 @@ decode_method_ref_with_target (MonoAotModule *module, MethodRef *ref, MonoMethod
 				if (!m)
 					return FALSE;
 				ref->method = mono_marshal_get_array_accessor_wrapper (m);
+			} else if (subtype == WRAPPER_SUBTYPE_UNSAFE_ACCESSOR) {
+				MonoMethod *m = decode_resolve_method_ref (module, p, &p, error);
+				MonoUnsafeAccessorKind kind = (MonoUnsafeAccessorKind) decode_value (p, &p);
+				uint32_t name_len = decode_value (p, &p);
+				const char *member_name = (const char*)p;
+				p += name_len + 1;
+				if (!m)
```

Comment:
Shouldn't we check for this immediately? Seems odd to do work prior to making this type of check.

Reply:
yea, I think when we were working on this, I thought it would be important to read the whole record for this wrapper subtype before returning.  But `p` is local (the out-var is `endbuf` which is assigned all the way at the ned of the function), so it doesn't actually matter.

This check can be moved up immediately after decoding the method ref.

Reply:
Make sense. Changed.